src/tlo/methods/healthburden.py

```diff
@@ -643,15 +710,69 @@ def apply(self, population):
 
         # Merge into a dataframe with the correct multi-index (the multi-index from combine is subtly different)
         self.module.years_lived_with_disability = \
-            pd.DataFrame(index=self.module.multi_index_for_age_and_wealth_and_time)\
-              .merge(combined, left_index=True, right_index=True, how='left')
+            pd.DataFrame(index=self.module.multi_index_for_age_and_wealth_and_time) \
+                .merge(combined, left_index=True, right_index=True, how='left')
 
         # Check multi-index is in check and that the addition of DALYS has worked
         assert self.module.years_lived_with_disability.index.equals(self.module.multi_index_for_age_and_wealth_and_time)
         assert abs(self.module.years_lived_with_disability.sum().sum() - (dalys_to_add + dalys_current)) < 1e-5
         self.module.check_multi_index()
 
 
+class Get_Current_Prevalence(RegularEvent, PopulationScopeEventMixin):
+    """
+    This event runs every month and asks each disease module to report the prevalence of each disease
+    during the previous month.
+    """
+
+    def __init__(self, module):
+            super().__init__(module, frequency=DateOffset(months=1))
+
+    def apply(self, population):
+        if not self.module.recognised_modules_names or not self.module.causes_of_disability:
+            return
+
+        # Calculate the population size
+        population_size = len(self.sim.population.props[self.sim.population.props['is_alive']])
+
+        # Create a DataFrame with one row and assign the population size
+        prevalence_from_each_disease_module = pd.DataFrame({'population': [population_size]})
+        for disease_module_name in self.module.recognised_modules_names:
+            if disease_module_name in ['NewbornOutcomes', 'PostnatalSupervisor', 'DiseaseThatCausesA',
+                                       'ChronicSyndrome']:
+                continue
+
+            disease_module = self.sim.modules[disease_module_name]
+            prevalence_from_disease_module = disease_module.report_prevalence()
+
+            if disease_module_name == "CardioMetabolicDisorders":
```

Comment:
as above, rather than be a special case, can we insetad detect if the thing being returned in a pd.DataFrame or a pd.Series and take it from there (e.g. if instance(prevalence_from_disease_module, pd.DataFrame))...)